load-gen/src/main/java/io/smartcat/data/loader/api/Terminateable.java

```diff
@@ -0,0 +1,13 @@
+package io.smartcat.data.loader.api;
+
+/**
+ * Has ability to terminate itself.
+ */
+public interface Terminateable {
```

Comment:
It does not create components, it just depends on them. DataSource, Worker and RateGenerator are provided to it.